app/src/main/java/com/cmput301f20t21/bookfriends/ui/library/owned/BorrowedOwnedDetailActivity.java

```diff
@@ -0,0 +1,60 @@
+package com.cmput301f20t21.bookfriends.ui.library.owned;
+
+import android.content.Intent;
+import android.os.Bundle;
+import android.view.View;
+import android.widget.Button;
+import android.widget.Toast;
+
+import androidx.annotation.Nullable;
+import androidx.lifecycle.ViewModelProvider;
+
+import com.cmput301f20t21.bookfriends.R;
+import com.cmput301f20t21.bookfriends.enums.REQUEST_STATUS;
+import com.cmput301f20t21.bookfriends.enums.SCAN_ERROR;
+import com.cmput301f20t21.bookfriends.ui.component.BaseDetailActivity;
+import com.cmput301f20t21.bookfriends.ui.scanner.ScannerBaseActivity;
+
+public class BorrowedOwnedDetailActivity extends BaseDetailActivity {
+    public static final int GET_SCANNED_ISBN = 2001;
+    private Button actionButton;
+    private BorrowedOwnedDetailViewModel vm;
+
+    @Override
+    protected void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        vm = new ViewModelProvider(this).get(BorrowedOwnedDetailViewModel.class);
+        actionButton = findViewById(R.id.detail_action_button);
+
+        vm.getRequest(book).observe(this, request -> {
+            if (request.getStatus().equals(REQUEST_STATUS.RETURNING)) {
+                actionButton.setText(R.string.scan_receive);
+                actionButton.setOnClickListener(this::openScanner);
+            }
```

Comment:
can you add handling to two more status here? mainly REQUEST_STATUS.BORROWED and REQUEST_STATUS.CLOSED
for BORROWED currently just disable the view of the button (we might implement something later if we have time)
for CLOSED set the button text to "Successfully got back [BookTitle]"

An example can be found in `com.cmput301f20t21.bookfriends.ui.borrow.accepted.AcceptedDetailAcitivty`'

![image](https://user-images.githubusercontent.com/43459469/100147745-007b9c00-2e59-11eb-95d0-cf99793e9005.png)